starter-code/database.js

```diff
@@ -133,8 +146,16 @@ class Database {
         // productName is the name of the product that we want to buy
         // Think if you may need to implement two queries chained
         // remeber once it's finish to comment callback('Error buying product');
-        
-        callback('Error buying product');
+        database.collection("products").find({"name": productName }).toArray((error, products) => {
+
+          database.collection("users").updateOne(
+            { "firstName": userFirstName },
+            { $push: {"shoppingCart": products[0] }}, null);
+          callback(null);
```

Comment:
Remember that queries are asynchronous. You wrote the `callback` function call right after the query statement. You have to guarantee that the query succeeded in the meantime, so you should rather have passed the callback to `updateOne`.